profiling/variorum-connector/variorum-connector.cpp

```diff
@@ -357,25 +508,99 @@ void kokkosp_init_library(const int loadSeq, const uint64_t interfaceVer,
 }
 
 void kokkosp_finalize_library() {
-  if (usingMPI) {
-    variorum_call_mpi();
-  } else {
-    variorum_call();
+  /*	std::cout << "kokkos library call\n" << std::endl;
+    if (usingMPI) {
+      variorum_call_mpi();
+    } else {
+     variorum_call();
+    }
+    time_t total_time;
+    time_t end_time;
+    time(&end_time);
+    std::cout << "End Time: " << end_time << "\nStart Time: " << start_time
+              << "\n";
+    total_time = end_time - start_time;
+
+    std::cout << "The kokkos library was alive for " << total_time << "
+    seconds."
+              << std::endl;*/
+}
+
+std::ostream &operator<<(
+    std::ostream &os,
+    const Kokkos::Tools::Experimental::DeviceType &deviceType) {
+  switch (deviceType) {
+    case Kokkos::Tools::Experimental::DeviceType::Serial:
+      os << "CPU";
+      output += "Type: CPU";
+      break;
+    case Kokkos::Tools::Experimental::DeviceType::OpenMP:
+      os << "OpenMP";
+      output += "Type: OpenMP";
+      break;
+    case Kokkos::Tools::Experimental::DeviceType::Cuda:
+      os << "cuda";
+      output += "Type: CUDA";
+      break;
+    case Kokkos::Tools::Experimental::DeviceType::HIP:
+      os << "hip";
+      output += "Type: HIP";
+      break;
+    case Kokkos::Tools::Experimental::DeviceType::OpenMPTarget:
+      os << "openmptarget";
+      output += "Type: openmptarget";
+      break;
+    case Kokkos::Tools::Experimental::DeviceType::HPX:
+      os << "hpx";
+      output += "Type: hpx";
+      break;
+    case Kokkos::Tools::Experimental::DeviceType::Threads:
+      os << "threads";
+      output += "Type: threads";
+      break;
+    case Kokkos::Tools::Experimental::DeviceType::SYCL:
+      os << "sycl";
+      output += "Type: SYCL";
+      break;
+    case Kokkos::Tools::Experimental::DeviceType::OpenACC:
+      os << "openacc";
+      output += "Type: OPENACC";
+      break;
+
+    default:
+      os << "Unknown Device Type";
+      output += "Type: Uknown Device Type";
+      break;
   }
-  time_t total_time;
-  time_t end_time;
-  time(&end_time);
-  std::cout << "End Time: " << end_time << "\nStart Time: " << start_time
-            << "\n";
-  total_time = end_time - start_time;
-
-  std::cout << "The kokkos library was alive for " << total_time << " seconds."
-            << std::endl;
+  return os;
```

Comment:
Define a function such as https://github.com/kokkos/kokkos-tools/blob/5e895a36f40bfce363c1424a6f5529a9ef6b64de/profiling/space-time-stack/kp_space_time_stack.cpp#L54-L67 instead. We don't want to make `Kokkos::Tools::Experimental::DeviceType` implicitly printable.